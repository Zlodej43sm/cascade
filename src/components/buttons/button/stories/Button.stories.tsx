import React from 'react';
import { Meta, Story } from '@storybook/react';

import { buttonArguments } from '../__mocks__/Button.mock';
import Button, { ButtonProps } from '../Button';

export default {
  title: 'Buttons/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => (<Button {...args} />);

export const Default = Template.bind({});
Default.args = buttonArguments.defaultProps;

export const DefaultCompact = Template.bind({});
DefaultCompact.args = buttonArguments.defaultCompactProps;

export const DefaultDisabled = Template.bind({});
DefaultDisabled.args = buttonArguments.defaultDisabledProps;

export const Secondary = Template.bind({});
Secondary.args = buttonArguments.secondaryProps;

export const SecondaryCompact = Template.bind({});
SecondaryCompact.args = buttonArguments.secondaryCompactProps;
