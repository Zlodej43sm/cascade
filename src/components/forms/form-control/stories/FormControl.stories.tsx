import React from 'react';
import { Meta, Story } from '@storybook/react';

import FormControl, { FormControlProps } from '../FormControl';

export default {
  title: 'Forms/FormControl',
  component: FormControl,
} as Meta;

const Template: Story<FormControlProps> = (args) => (<FormControl {...args} />);

export const Default = Template.bind({});
Default.args = {
  label: 'Label',
  children: 'children',
};

export const NoLabel = Template.bind({});
NoLabel.args = {
  children: 'children',
};
