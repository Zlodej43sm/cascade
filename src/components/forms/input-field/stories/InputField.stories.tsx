import React from 'react';
import { Meta, Story } from '@storybook/react';

import InputField, { InputFieldProps } from '../InputField';

export default {
  title: 'Forms/InputField',
  component: InputField,
} as Meta;

const Template: Story<InputFieldProps> = (args) => (<InputField {...args} />);

export const Default = Template.bind({});
Default.args = {
  name: 'test-field',
  label: 'Label',
};
