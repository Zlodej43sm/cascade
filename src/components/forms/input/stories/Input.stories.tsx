import React from 'react';
import { Meta, Story } from '@storybook/react';

import Input, { InputProps } from '../Input';

export default {
  title: 'Forms/Input',
  component: Input,
} as Meta;

const Template: Story<InputProps> = (args) => (<Input {...args} />);

export const Default = Template.bind({});
Default.args = {
  placeholder: 'Enter value',
};
