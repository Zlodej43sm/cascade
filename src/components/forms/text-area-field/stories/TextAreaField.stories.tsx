import React from 'react';
import { Meta, Story } from '@storybook/react';

import TextAreaField, { TextAreaFieldProps } from '../TextAreaField';

export default {
  title: 'Forms/TextAreaField',
  component: TextAreaField,
} as Meta;

const Template: Story<TextAreaFieldProps> = (args) => (<TextAreaField {...args} />);

export const Default = Template.bind({});
Default.args = {
  name: 'test-field',
  label: 'Label',
};
