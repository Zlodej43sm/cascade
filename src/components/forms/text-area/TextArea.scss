@use '../../../styles/colors';
@use '../../../styles/typography';

.input {
  @include typography.mulish(14px, 18px);
  padding: 8px;
  box-sizing: border-box;
  width: 100%;
  background-color: colors.$grey-10;
  border: none;
  border-radius: 5px;
  color: colors.$base-dark;
}
