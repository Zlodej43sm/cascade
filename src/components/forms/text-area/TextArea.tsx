import React from 'react';
import clsx from 'clsx';

import styles from './TextArea.scss';

export type TextAreaProps = Pick<React.HTMLProps<HTMLTextAreaElement>, 'id' | 'name' | 'value' | 'placeholder' | 'className' | 'onChange' | 'onBlur'>;

function TextArea({ className, ...rest }: TextAreaProps) {
  return (
    <textarea className={clsx(styles.input, className)} {...rest} />
  );
}

export default TextArea;
