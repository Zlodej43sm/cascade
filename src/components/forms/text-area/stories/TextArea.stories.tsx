import React from 'react';
import { Meta, Story } from '@storybook/react';

import TextArea, { TextAreaProps } from '../TextArea';

export default {
  title: 'Forms/TextArea',
  component: TextArea,
} as Meta;

const Template: Story<TextAreaProps> = (args) => (<TextArea {...args} />);

export const Default = Template.bind({});
Default.args = {
  placeholder: 'Enter value',
};
