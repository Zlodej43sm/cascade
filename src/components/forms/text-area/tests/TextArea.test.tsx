import React from 'react';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from '../stories/TextArea.stories';

const { Default } = composeStories(stories);

describe('TextArea', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<Default />);
    expect(asFragment()).toMatchSnapshot();
  });
});
