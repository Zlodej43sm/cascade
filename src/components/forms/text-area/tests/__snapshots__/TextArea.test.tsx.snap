// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Should match snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="input"
    placeholder="Enter value"
  />
</DocumentFragment>
`;
